src/pymor/bindings/scipy.py

```diff
@@ -313,6 +315,24 @@ def _solve_impl(self, matrix, V, initial_guess, promoted_type):
         return R
 
 
+class ScipyQRLSTSQSolver(ScipyLinearSolver):
+
+    least_squares = True
+    _qr_factors = WeakRefCache()
+
+    def _solve_impl(self, matrix, V, initial_guess, promoted_type):
+        try:
+            Q, R = self._qr_factors.get(matrix)
+        except KeyError:
+            try:
+                Q, R = qr(matrix, mode='economic', check_finite=self.check_finite)
+            except np.linalg.LinAlgError as e:
+                raise InversionError(f'{type(e)!s}: {e!s}') from e
+            self._qr_factors.set(matrix, (Q, R))
+        U = solve_triangular(R, Q.T @ V, lower=False)
```

Comment:
QR with column pivoting is better for ill-conditioned problems (and SVD is even better).